app/src/components/card/Card.tsx

```diff
@@ -57,7 +58,11 @@ function Card(
       <header id={headerId}>
         {collapsible ? (
           <button
-            onClick={() => setIsCollapsed(!isCollapsed)}
+            onClick={() => {
```

Comment:
This is probably fine but you can also use useEffect for this to synchronize onCollapse with collapse state